node/src/components/deploy_acceptor/tests.rs

```diff
@@ -0,0 +1,584 @@
+            | TestScenario::FromClientValidDeploy
+            | TestScenario::FromClientRepeatedValidDeploy => Source::Client,
+        }
+    }
+
+    fn deploy(&self, rng: &mut NodeRng) -> Deploy {
+        let mut deploy = Deploy::random(rng);
+        match self {
+            TestScenario::FromPeerInvalidDeploy | TestScenario::FromClientInvalidDeploy => {
+                deploy.invalidate()
+            }
+            TestScenario::FromPeerValidDeploy
+            | TestScenario::FromPeerRepeatedValidDeploy
+            | TestScenario::FromClientMissingAccount
+            | TestScenario::FromClientInsufficientBalance
+            | TestScenario::FromClientValidDeploy
+            | TestScenario::FromClientRepeatedValidDeploy
+            | TestScenario::FromPeerRegression => (),
+        }
+        deploy
+    }
+
+    fn is_from_peer(&self) -> bool {
+        match self {
+            TestScenario::FromPeerInvalidDeploy
+            | TestScenario::FromPeerValidDeploy
+            | TestScenario::FromPeerRepeatedValidDeploy
+            | TestScenario::FromPeerRegression => true,
+            TestScenario::FromClientInvalidDeploy
+            | TestScenario::FromClientMissingAccount
+            | TestScenario::FromClientInsufficientBalance
+            | TestScenario::FromClientValidDeploy
+            | TestScenario::FromClientRepeatedValidDeploy => false,
+        }
+    }
+
+    fn is_valid_deploy_case(&self) -> bool {
+        match self {
+            TestScenario::FromPeerRepeatedValidDeploy
+            | TestScenario::FromPeerValidDeploy
+            | TestScenario::FromClientRepeatedValidDeploy
+            | TestScenario::FromClientValidDeploy => true,
+            TestScenario::FromPeerInvalidDeploy
+            | TestScenario::FromPeerRegression
+            | TestScenario::FromClientInsufficientBalance
+            | TestScenario::FromClientMissingAccount
+            | TestScenario::FromClientInvalidDeploy => false,
+        }
+    }
+}
+
+struct Reactor {
+    storage: Storage,
+    deploy_acceptor: DeployAcceptor,
+    _storage_tempdir: TempDir,
+    test_scenario: TestScenario,
+}
+
+impl reactor::Reactor for Reactor {
+    type Event = Event;
+    type Config = TestScenario;
+    type Error = Error;
+
+    fn new(
+        config: Self::Config,
+        _registry: &Registry,
+        _event_queue: EventQueueHandle<Self::Event>,
+        _rng: &mut NodeRng,
+    ) -> Result<(Self, Effects<Self::Event>), Self::Error> {
+        let (storage_config, storage_tempdir) = storage::Config::default_for_tests();
+        let storage_withdir = WithDir::new(storage_tempdir.path(), storage_config);
+        let storage = Storage::new(
+            &storage_withdir,
+            None,
+            ProtocolVersion::from_parts(1, 0, 0),
+            false,
+            "test",
+        )
+        .unwrap();
+
+        let deploy_acceptor = DeployAcceptor::new(
+            super::Config::new(true),
```

Comment:
* Would it be reasonable to test the `false` variant as well? If I read the `deploy_acceptor.rs` correctly, we can have a deploy with invalid account which could still be accepted.

* Also, consider using a named const here: `super::Config::new(VERIFY_ACCOUNTS)` or similar

Reply:
No, it doesn't make sense to have a `false` variant cause that means it won't perform account verification, which is what we need to test here

Reply:
I agree with your point on the `VERIFY_ACCOUNTS` const

Reply:
Sure. I just observed that in the `deploy_acceptor.rs` we can replace the following line

`verify_accounts: config.verify_accounts()`

with

`verify_accounts: true`

and all `deploy_acceptor` tests still pass. If this is tested in another place (in gossiper, probably) though, I'm fine.

Reply:
Discussed with @rafal-ch , the `false` scenario contains a subset of code paths in the `true` case, i.e account verification